framework/tests/all-systems/TypeVars.java

```diff
@@ -3,7 +3,7 @@
 
     class Test1<T> {
         void m() {
-            @SuppressWarnings("unchecked")
+            @SuppressWarnings({"unchecked", "assignment.type.incompatible"})
```

Comment:
If it's just the units checker, we could suppress the warning only for it. A short comment might be helpful to avoid confusion why this is necessary.